Intersect.Client/MonoGame/Graphics/MonoRenderer.cs

```diff
@@ -615,33 +610,45 @@ public override GameTileBuffer CreateTileBuffer()
             var origin = Vector2.Zero;
             if (Math.Abs(rotationDegrees) > 0.01)
             {
-                rotationDegrees = (float) (Math.PI / 180 * rotationDegrees);
+                rotationDegrees = (float)(Math.PI / 180 * rotationDegrees);
                 origin = new Vector2(sw / 2f, sh / 2f);
 
-                //TODO: Optimize in terms of memory AND performance.
-                var pnt = new Pointf(0, 0);
-                var pnt1 = new Pointf((float) tw, 0);
-                var pnt2 = new Pointf(0, (float) th);
-                var cntr = new Pointf((float) tw / 2, (float) th / 2);
-
-                var pntMod = Rotate(pnt, cntr, rotationDegrees);
-                var pntMod2 = Rotate(pnt1, cntr, rotationDegrees);
-                var pntMod3 = Rotate(pnt2, cntr, rotationDegrees);
-
-                var width = (int) Math.Round(GetDistance(pntMod.X, pntMod.Y, pntMod2.X, pntMod2.Y));
-                var height = (int) Math.Round(GetDistance(pntMod.X, pntMod.Y, pntMod3.X, pntMod3.Y));
```

Comment:
This old code is better (more legible, less confusing), change it back.

The new code has no real performance benefit.

Reply:
Replaced the Pointf struct with separate float variables for the x and y coordinates becayse it saves memory (Pointf is a struct and requires more memory than two separate float variables). Also modified the Rotate method to accept the coordinates as ref variables rather than as a Pointf struct. This allows the method to directly modify the original variables **rather than creating new ones**, which _can_ save memory and improve performance. As for cntr, it was unnecessary to create it as Pointf variable under this scenario so i replaced it with two separate float variables for the x and y coordinates. This also _should_ save some memory. i essentially tried to solve the todo from the comment within the code here.

Reply:
> Pointf is a struct and requires more memory than two separate float variables

Pointf _is_ a struct. That is exactly why it _**does not**_ require more memory than two separate float variables.

All value types (structs, enums, non-string primitives) have _no_ additional overhead.

_Classes_ have additional memory consumption because of the overhead of inheritance (i.e. pointers to vtables for overloads, etc).

Reply:
I'll change it back then 👍🏽